executor/include/execution/trait/can_require.hpp

```diff
@@ -0,0 +1,33 @@
+//
+// Created by Shrijit Singh on 14/06/20.
+//
+
+#pragma once
+
+#include <type_traits>
+
+namespace execution {
+
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor> &&
+                  Property::is_requirable &&
+                  Property::template static_query<Executor>(),
+              int> = 0>
+Executor require(Executor ex, const Property t) {
+  return ex.require(t);
+}
+
+template <typename Executor, typename Properties, typename = void>
+struct can_require : std::false_type {};
+
+template <typename Executor, typename Property>
+struct can_require<Executor, Property,
+                   std::void_t<decltype(execution::require(
+                       std::declval<Executor>(), std::declval<Property>()))>>
```

Comment:
Maybe explain what you're checking: `require(executor, property)`

Reply:
Got it, will add some documentation, once the API is more stable.